docs/index.md

```diff
@@ -56,24 +56,95 @@ For the best experience using the SAP BTP provider, we recommend applying the co
+
+
+### Authentication to SAP BTP using Username and Password
+
+A supported authentication flow is the *username-password flow*. The configuration of the provider using the SAP default identity provider requires values for the `globalaccount`, `username` and `password`.
+
+An example for a provider configuration looks like this:
+
+```terraform
+# Configure the BTP Provider
+# We recommend to source username and password from environment variables
+provider "btp" {
+  globalaccount = "my-global-account-subdomain"
+  username      = var.username
+  password      = var.password
+}
+```
+
+The configuration of the provider using a custom identity provider requires in addition to the previous example a value for the parameter `idp`.
+
+An example for a provider configuration with a custom identity provider looks like this:
+
+```terraform
+# Configure the BTP Provider
+# We recommend to source username and password from environment variables
+provider "btp" {
+  globalaccount = "my-global-account-subdomain"
+  idp           = "customerTenant.accounts.ondemand.com"
+  username      = var.username
+  password      = var.password
+}
+```
+
+We recommend sourcing the username and password via the environment variables `BTP_USERNAME` and `BTP_PASSWORD`. In case you want to provide the username and password via variables make sure to follow the guidance given in the [Hashicorp documentation](https://developer.hashicorp.com/terraform/tutorials/configuration-language/sensitive-variables) and never commit the values to a source code management system.
+
+### Authentication to SAP BTP using X509 certificates
+
+A supported authentication flow is the *X509 flow*. This authentication must be configured in the SAP Cloud Identity Service as described in [help.sap.com](https://help.sap.com/docs/cloud-identity-services/cloud-identity-services/configure-x-509-client-certificates-for-user-authentication).
+
+For our example configuration we assume that the X509 certificate and the X509 private key are stored in local files called `btp-tech-user-cert.pem` and `btp-tech-user-key.pem`. The files must be PEM encoded. You can achieve this e.g., via [openssl](https://docs.openssl.org/master/man1/openssl-x509/).
+
+An example for a provider configuration looks like this:
+
+```terraform
+# Configure the BTP Provider
+provider "btp" {
+  globalaccount          = "my-global-account-subdomain"
+  idp                    = "customerTenant.accounts.ondemand.com"
+  username               = "btp-tech-user@customer-domain.de"
+  tls_client_certificate = file("PATH_TO_FILE/btp-tech-user-cert.pem")
+  tls_client_key         = file("PATH_TO_FILE/btp-tech-user-key.pem")
+  tls_idp_url            = "https://customerTenant.accounts.ondemand.com"
+}
```

Comment:
Although it can be implicit, can we add a note stating that when using the `x509` approach the IDP is required?